src/query/sql/src/planner/plans/join.rs

```diff
@@ -434,23 +441,35 @@ fn evaluate_by_histogram(
     left_hist: &Histogram,
     right_hist: &Histogram,
     new_ndv: &mut Option<f64>,
+    new_min: &Option<Datum>,
+    new_max: &Option<Datum>,
 ) -> Result<f64> {
     let mut interleaved_buckets = vec![];
     let mut all_ndv = 0.0;
-    for (idx, (left_bucket, right_bucket)) in left_hist
-        .buckets
-        .iter()
-        .zip(right_hist.buckets.iter())
-        .enumerate()
+    // Use new_min/new_max to prune buckets
+    let (left_buckets, right_buckets) = prune_buckets(left_hist, right_hist, new_min, new_max)?;
+
+    for (idx, (left_bucket, right_bucket)) in
+        left_buckets.iter().zip(right_buckets.iter()).enumerate()
```

Comment:
After calling `prune_buckets`, the lengths of `left_buckets` and `right_buckets` may be different. Can this one-dimensional `for` still generate accurate results?

Reply:
No accurate results are here, just an estimate.